### PR TITLE
Add tests, docs for otelcol container args

### PR DESCRIPTION
**Pull request by VineethReddy02**
Add unit, e2e tests for args that are passed from otelcol CR spec to otelcol container args.

fixes: #12

### CODE REVIEW

File: docs/otelcol_cr_spec.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# OpenTelemetryCollector Custom Resource Specification
+
+The below `OpenTelemetryCollector` custom resource contains all the specification that can be configured. 
+
+```
+apiVersion: opentelemetry.io/v1alpha1
```
Comment (jpkrohling) [inline docs/otelcol_cr_spec.md:6]: Is this auto-generated? Will we need to update this manually whenever the CRD changes?

Comment (VineethReddy02) [reply docs/otelcol_cr_spec.md:6]: This is a manually created doc. I didn't find any effective way of generating docs for CRD spec. :/

Comment (jpkrohling) [reply docs/otelcol_cr_spec.md:6]: Alright, I'll merge this one as it's an improvement already, but could you please open a new issue to figure out how to generate docs out of the CRDs?

**Review by jpkrohling — APPROVED** [review]

File: docs/otelcol_cr_spec.md (hunk 1 above)
Comment (VineethReddy02) [reply docs/otelcol_cr_spec.md:6]: Created a new issue on generating docs from CRD spec [here](https://github.com/open-telemetry/opentelemetry-operator/issues/325)

**Pull request merged**